File: scripts/hbase/example_1.java
import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.hbase.*;
import org.apache.hadoop.hbase.client.*;
import org.apache.hadoop.hbase.util.Bytes;

public class HBaseExample {
    public static void main(String[] args) throws Exception {
        // Configuration
        Configuration config = HBaseConfiguration.create();
        config.set("hbase.zookeeper.quorum", "localhost");
        config.set("hbase.zookeeper.property.clientPort", "2181");

        // Connection
        try (Connection connection = ConnectionFactory.createConnection(config)) {

            // Get table
            Table table = connection.getTable(TableName.valueOf("user_activity"));

            // Put operation
            Put put = new Put(Bytes.toBytes("user003"));
            put.addColumn(
                Bytes.toBytes("profile"),
                Bytes.toBytes("name"),
                Bytes.toBytes("Charlie Brown")
            );
            table.put(put);

            // Get operation
            Get get = new Get(Bytes.toBytes("user003"));
            Result result = table.get(get);
            byte[] value = result.getValue(
                Bytes.toBytes("profile"),
                Bytes.toBytes("name")
            );
            System.out.println("Name: " + Bytes.toString(value));

            // Scan operation
            Scan scan = new Scan();
            scan.addFamily(Bytes.toBytes("profile"));
            ResultScanner scanner = table.getScanner(scan);

            for (Result res : scanner) {
                System.out.println("Row: " + Bytes.toString(res.getRow()));
            }
            scanner.close();
            table.close();
        }
    }
}

File: scripts/hbase/example_1.py
import happybase

# Connect to HBase
connection = happybase.Connection('localhost')

# Open table
table = connection.table('user_activity')

# Put data
table.put(b'user004', {
    b'profile:name': b'Diana Prince',
    b'profile:email': b'diana@example.com',
    b'events:login_2024-01-17': b'08:00:00'
})

# Get data
row = table.row(b'user004')
print(f"Name: {row[b'profile:name'].decode()}")

# Scan table
for key, data in table.scan():
    print(f"Row: {key.decode()}")
    for col, val in data.items():
        print(f"  {col.decode()} => {val.decode()}")

# Delete
table.delete(b'user004')

connection.close()